x/kavadist/keeper/mint.go

```diff
@@ -0,0 +1,49 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cdptypes "github.com/kava-labs/kava/x/cdp/types"
+	"github.com/kava-labs/kava/x/kavadist/types"
+)
+
+// MintPeriodRewards mints new tokens according to the inflation schedule specified in the paramters
+func (k Keeper) MintPeriodRewards(ctx sdk.Context) sdk.Error {
+	params := k.GetParams(ctx)
+	if !params.Active {
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeKavaDist,
+				sdk.NewAttribute(types.AttributeKeyStatus, types.AttributeValueInactive),
+			),
+		)
+		return nil
+	}
+	previousBlockTime, found := k.GetPreviousBlockTime(ctx)
+	if !found {
+		previousBlockTime = ctx.BlockTime()
+		k.SetPreviousBlockTime(ctx, previousBlockTime)
+		return nil
+	}
+	timeElapsed := sdk.NewInt(ctx.BlockTime().Unix() - previousBlockTime.Unix())
```

Comment:
This doesn't take into account the period start/end times when calculating timeElapsed. So the newly minted coins for a period could be up to one block's worth different from the "ideal" amount. It could catch us out if the last blocktime is really long in the case of a chain halt. Probably worth a comment or fix.